syntax = "proto3";

package demo;

service DemoService {
  // Unary RPC
  rpc UnaryCall (Request) returns (Response);
  // Server streaming RPC
  rpc ServerStreamingCall (Request) returns (stream Response);
  // Client streaming RPC
  rpc ClientStreamingCall (stream Request) returns (Response);
  // Bidirectional streaming RPC
  rpc BidiStreamingCall (stream Request) returns (stream Response);
}

message Request {
  string message = 1;
}

message Response {
  string message = 1;
}
